src/loaders/ncbi_downloader/ncbi_downloader.py

```diff
@@ -79,3 +222,70 @@ def download_genome_files(gene_ids: list[str], target_file_ext: list[str], resul
         print(f'Failed to download {failed_ids}')
 
     return failed_ids
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='PROTOTYPE - Download genome files from NCBI FTP server.')
+
+    gtdb_release_vers = _parse_gtdb_release_vers()  # available GTDB release versions
+
+    required = parser.add_argument_group('required named arguments')
+    optional = parser.add_argument_group('optional arguments')
+
+    # Required flag argument
+    required.add_argument('--download_file_ext', required=True, type=str, nargs='+',
+                          help='Download only files that match given extensions.')
+    required.add_argument('--release_ver', required=True, type=str, choices=gtdb_release_vers,
+                          help='GTDB release version')
+
+    # Optional argument
+    optional.add_argument('--root_dir', type=str, default='/global/cfs/cdirs/kbase/collections/sourcedata',
+                          help='Root directory. (default: /global/cfs/cdirs/kbase/collections/sourcedata)')
+    optional.add_argument('--source', type=str, default='GTDB',
+                          help='Source of data (default: GTDB)')
+    optional.add_argument('--threads', type=int,
+                          help='Number of threads. (default: half of system cpu count)')
+    optional.add_argument('--overwrite', action='store_true',
+                          help='Overwrite existing files.')
+
+    args = parser.parse_args()
+
+    (download_file_ext,
+     release_ver,
+     root_dir,
+     source,
+     threads,
+     overwrite) = (args.download_file_ext, args.release_ver, args.root_dir, args.source,
+                   args.threads, args.overwrite)
+
+    if source not in SOURCE:
+        raise ValueError(f'Unexpected source. Currently supported sources: {SOURCE}')
+
+    if release_ver not in gtdb_release_vers:
+        raise ValueError(f'Unsupported GTDB release version: {release_ver}')
```

Comment:
This is unreachable isn't it? Argparse will catch bad release vers

Reply:
yup. removed.

Reply:
:+1: